### PR TITLE
[iOS] Shell flyout header on iOS now is dynamic based on the header content

### LINKED ISSUE #3875: [Shell] Flyout menu items should start below my header, no matter how tall
https://github.com/xamarin/Xamarin.Forms/blob/ba229c0be7996e2604b5dd72a9a1edaf1daa4e3c/Xamarin.Forms.Platform.iOS/Renderers/ShellTableViewController.cs#L12

When my header is taller than 200, the content overlaps the menu. Please position the menu below the header always. 

screenshot 2018-09-22 17 24 23

```<?xml version="1.0" encoding="UTF-8"?>
<Shell xmlns="http://xamarin.com/schemas/2014/forms" 
       xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
       xmlns:views="clr-namespace:Movies.Views"
       xmlns:converters="clr-namespace:Movies.Converters;assembly=Movies"
       RouteScheme="https"
       RouteHost="movies.com"
       FlyoutHeaderBehavior="CollapseOnScroll"
       FlyoutBackgroundColor="{StaticResource MenuBackgroundColor}"
       x:Class="Movies.Shell">
    
    <Shell.Resources>
        <ResourceDictionary>

            <!-- GENERAL COLORS -->
            <Color x:Key="WhiteColor">#FFFFFF</Color>
            <Color x:Key="BlackColor">#000000</Color>

            <!-- THEME COLORS -->
            <Color x:Key="AccentColor">#F5C210</Color>
            <Color x:Key="ToolbarColor">#151C25</Color>
            <Color x:Key="BackgroundColor">#1E2634</Color>
            <Color x:Key="DarkBackgroundColor">#151C25</Color>
            <Color x:Key="MenuBackgroundColor">#44545C</Color>
            
            <!-- CONVERTERS -->
            <converters:InverseBoolConverter x:Key="InverseBoolConverter" />
            <converters:MenuItemTypeConverter x:Key="MenuItemTypeConverter" />
            <converters:MovieImageConverter x:Key="MovieImageConverter" /> 
            <converters:ReleaseDateConverter x:Key="ReleaseDateConverter" />
            <converters:ToUpperConverter x:Key="ToUpperConverter" />
            
        </ResourceDictionary>
    </Shell.Resources>
    
    <Shell.FlyoutHeader>
        <ContentView HeightRequest="600">
            <StackLayout BackgroundColor="Black" Orientation="Vertical">
                <Image Source="https://sec.ch9.ms/sessions/connect/T128.png" 
                       HorizontalOptions="Fill"
                       VerticalOptions="Start"
                       Aspect="AspectFill"/>
                <StackLayout Spacing="6" Padding="8">
                    <Label Text="David Ortinau" TextColor="White"></Label>
                    <Label Text="david.ortinau@microsoft.com" TextColor="White"></Label>
                </StackLayout>
            </StackLayout>
        </ContentView>
    </Shell.FlyoutHeader>
    
    
    <ShellItem Title="Discover" Route="discover" FlyoutIcon="movies_search.png">
        <ShellContent ContentTemplate="{DataTemplate views:HomeView}"  Icon="movies_search.png"/>
    </ShellItem>
    <ShellItem Title="Movies" Route="movies" FlyoutIcon="movies_movie.png">
        <ShellContent ContentTemplate="{DataTemplate views:MoviesView}"/>
    </ShellItem>
    <ShellItem Title="TV Shows" Route="shows" FlyoutIcon="movies_shows.png">
        <ShellContent ContentTemplate="{DataTemplate views:ShowsView}"/>
    </ShellItem>
    <ShellItem Title="Upcoming" Route="upcoming" FlyoutIcon="movies_upcoming.png">
        <ShellContent ContentTemplate="{DataTemplate views:UpcomingView}"/>
    </ShellItem>
    <ShellItem Title="Favourites" Route="favourites" FlyoutIcon="movies_favourites.png">
        <ShellContent ContentTemplate="{DataTemplate views:MoviesView}"/>
    </ShellItem>
    <ShellItem Title="Settings" Route="settings" FlyoutIcon="movies_settings.png">
        <ShellContent ContentTemplate="{DataTemplate views:HomeView}"/>
    </ShellItem>
    
</Shell>
```
Parent: #2415

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
- Made it so the content height gets measured out
- changed minsize from 44 to 56 to match android

### Issues Resolved ### 
- fixes #3875

### Platforms Affected ### 
- iOS

### Testing Procedure ###
I updated store shells height to be a number that's not 200 (which was the previous default) so you can use store shell to verify android and iOS are the same size. Also click on the different FH scroll behaviors

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

**Comment by PureWeen** [comment]
@paymicro great suggestions and thanks for adding the slider. I'm pretty sure I've addressed all the things you brought up.

File: Xamarin.Forms.Platform.iOS/Renderers/UIContainerView.cs [hunk 1]
```diff
@@ -17,11 +19,40 @@ public UIContainerView(View view)
 			Platform.SetRenderer(view, _renderer);
 
 			AddSubview(_renderer.NativeView);
+			ClipsToBounds = true;
+			view.MeasureInvalidated += OnMeasureInvalidated;
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/UIContainerView.cs]: We need to unsubscribe on Dispose

**Comment by melucas** [comment]
Was this fix included in Pre10?  If so, things seem to be not right.

There is now a gap/space between the top of the flyout header and the content.
Setting the padding or margins does not help.

How do you change that gap, or how can I get the flyout header content back at the top of the flyout as is Pre9 and earlier?

Image-1

**Comment by PureWeen** [comment]
@melucas this particular PR wasn't on pre10 but your issue might still be present

Do you have a repro by chance I can test? What device are you using in your screen shot there

**Comment by melucas** [comment]
@PureWeen,

I forgot to include that is in iOS (12.2) screen shot.  I can not share this repo, but I will try to create one with reproduces the problem.  The gap is new in pre 10.  It was not there in pre 9 an earlier.

**Comment by PureWeen** [comment]
@melucas what's the device in the pic? iphone8 ? iPhone 8 plus?

**Comment by melucas** [comment]
@PureWeen 

iPhone 6 Plus running iOS 12.2

**Comment by melucas** [comment]
Getting closer, it has something to do with the space for the iOS status bar.  It is the same height as that.

**Comment by melucas** [comment]
@PureWeen,
Attached is a freshly built iOS app which shows the issue.

If you down grade the project to Pre 9, the white bar at the top of the flyout is not there.

Second issue this show is that the menu items are not respecting the space the top header uses.  It my opinion, it should push the menu items down so there is no overlap.

[FlyoutDemo.zip](https://github.com/xamarin/Xamarin.Forms/files/3168354/FlyoutDemo.zip)

**Comment by PureWeen** [comment]
> Second issue this show is that the menu items are not respecting the space the top header uses. It my opinion, it should push the menu items down so there is no overlap.

So this PR is addressing this issue. With the 4.0 release the size of the header will be respected more

> If you down grade the project to Pre 9, the white bar at the top of the flyout is not there.

The behavior for this one (unfortunately) is still the same.

You can see the PR and justifications for the current state here
https://github.com/xamarin/Xamarin.Forms/pull/5990

But we are working on a better api to let users modulate the padding.

**Comment by melucas** [comment]
Thank you @PureWeen.

Back to Pre 9 for now.

Michael

**Comment by PureWeen** [comment]
[release.zip](https://github.com/xamarin/Xamarin.Forms/files/3245407/release.zip)

@melucas if you want to test out offsetting the header view to 0,0 here's a nuget

https://github.com/xamarin/Xamarin.Forms/pull/6383 has a description of how it all works together

Basically you just have to set the Margin on the view to 0,0 and then it will show at the top left corner

**Pull request merged**